src/restapi.rs

```diff
@@ -1,9 +1,27 @@
-use goose::goose::{GooseUser, TransactionResult};
+use goose::goose::{GooseUser, TransactionError, TransactionResult};
 use serde_json::json;
 use urlencoding::encode;
 
 use crate::utils::DisplayVec;
 
+pub async fn upload_advisories_and_delete_advisories(
```

Comment:
I think this should be multiple individual steps in goose. Assuming that's possible.